csharp/Adapter/Microsoft.Spark.CSharp/Interop/Ipc/JvmObjectReference.cs

```diff
@@ -21,6 +21,7 @@ public JvmObjectReference(string jvmReferenceId)
         {
             Id = jvmReferenceId;
             creationTime = DateTime.UtcNow;
+            WeakObjectManager.GetWeakObjectManager().AddWeakRefereceObject(this);
```

Comment:
Oh, I missed the interface. I think it would help to keep the interface and use it in unit tests. Add unit tests and see if you can remove the interface. If you can, go ahead and remove it to implement the suggestion above. If you cannot remove the interface, without impacting unit tests, just keep it.